02-prompting-101/improving-chatGPTs-responses.md

```diff
@@ -0,0 +1,518 @@
+
+> Compose a 250 character or less social media post about NeoAnthroNews.com. NeoAnthroNews.com is a brand new website that shares breaking news in the field of anthropology as well as stories of current and past anthropologists.
+
+With our added context, ChatGPT had everything it needed for a short message about our website. Generative tools can get things like word count wrong, so if there are important details like length, we should confirm if those constraints were respected. 
+
+<details>
+   <summary>Try out the updated prompt, then expand this section to see our final result.</summary>
+
+> "Dive into the rich tapestry of humanity with @NeoAnthroNews! Your go-to source for breaking anthropology news and inspiring stories of past and present anthropologists. 🌍✨ Explore the evolution of our world at NeoAnthroNews.com! #Anthropology #BreakingNews"
+
+Much better! The topics are correct, and we have some helpful hashtags. It turns out that the message doesn't meet all our needs though, it's 257 characters long, which breaks our request that the message be 250 characters or less. We could try regenerating the response or adjusting our wording to get ChatGPT to respect all our constraints.
+
+</details>  
+
+## Giving Examples/Shot Prompting
+
+As we’ve learned, generative AIs act very much like predictive text, but with a lot more information to use to make their predictions. No matter how much data they have though, there will always be specific tasks or response formats they they will not have encountered. 
+
+Especially when you need the output formatted a particular way or in a way that is difficult to describe, we can improve an AI Tool’s response by including examples (called “Shots” when talking about prompt engineering) of what it is that we want the AI tool to do. When we include a single example, this is called “Single Shot” prompting, and when we give it 2 or more examples it gets called “Few Shot” prompting.
+
+Let's say that we moderate a forum and want to categorize feedback from our users. We're going start from a zero-shot prompt and iterate on it until we have a few-shot prompt. For our example, we have a list of feedback like:
+
+> Feedback 1: I love this site but I can’t log in on iPhone!  
+> Feedback 2: The custom icons are so cute.  
+> Feedback 3: The cursor disappears after clicking the logo.  
+
+and as a response we want to get back:
+
+> Feedback 1: "I love this site but I can’t log in on iPhone!"  
+> Sentiment: negative  
+> Themes: iPhone login  
+> 
+> Feedback 2: "The custom icons are so cute."  
+> Sentiment: positive  
+> Themes: custom icons  
+> 
+> Feedback 3: "The cursor disappears after clicking the logo."  
+> Sentiment: negative  
+> Themes: UI, cursor  
+
+If we include no examples, no instructions, and simply submit our list of feedback as a prompt, generative AI tools like ChatGPT will still try to provide a helpful response, likely about how to address these pieces of user feedback. 
```

Comment:
That's exactly what's happening, the prompt is literally just the feedback. Would it be helpful if this paragraph went up before we show the prompt to be really clear about that?

Reply:
I'm thinking above the feedback when introducing the example, we could start the paragraph with something like "We're going to look at a prompt that's different from other's we've seen because we're not going to include _any_ direct instructions on what the generative AI should do. Let's say that we moderate a forum..."